build/autotools/ReadCommandLineArguments.m4

```diff
@@ -114,3 +114,16 @@ AC_ARG_ENABLE([tests],
                               [Build MongoDB C Driver tests.])],
               [],
               [enable_tests=yes])
+
+AC_MSG_CHECKING([whether to use system crypto profile])
+AC_ARG_ENABLE(system-crypto-profile,
+    AC_HELP_STRING([--enable-system-crypto-profile], [use system crypto profile [default=no]]),
+    [],
+    [enable_system_crypto_profile="no"])
+AC_MSG_RESULT([$enable_coverage])
+if test "x$enable_system_crypto_profile" = xyes; then
+    AC_SUBST(ENABLE_SYSTEM_CRYPTO_PROFILE, 1)
+else
+    AC_SUBST(ENABLE_SYSTEM_CRYPTO_PROFILE, 0)
+fi
+
```

Comment:
I'd prefer to see this in CheckSSL, and some sanity message "only affects openssl" and fail when not using openssl